src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3522,6 +3486,85 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 							}
 							break;
 
+						//
+						// static GetRuntimeEvent (this Type type, string name)
+						// static GetRuntimeField (this Type type, string name)
+						// static GetRuntimeMethod (this Type type, string name, Type[] parameters)
+						// static GetRuntimeProperty (this Type type, string name)
+						//
+						case var getRuntimeMember when (getRuntimeMember.StartsWith ("GetRuntime")
+							&& calledMethod.DeclaringType.Namespace == "System.Reflection"
+							&& calledMethod.DeclaringType.Name == "RuntimeReflectionExtensions"
+							&& calledMethod.Parameters.Count >= 2
+							&& calledMethod.Parameters [0].ParameterType.FullName == "System.Type"
+							&& calledMethod.Parameters [1].ParameterType.FullName == "System.String"): {
+
+								reflectionContext.AnalyzingPattern ();
+								DynamicallyAccessedMemberKinds? memberKind = null;
+								BindingFlags bindingFlags = BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public;
+
+								switch (getRuntimeMember) {
+									case var mt when getRuntimeMember.EndsWith ("Event"):
+										memberKind = DynamicallyAccessedMemberKinds.PublicEvents;
+										break;
+
+									case var mt when getRuntimeMember.EndsWith ("Field"):
+										memberKind = DynamicallyAccessedMemberKinds.PublicFields;
+										break;
+
+									case var mt when getRuntimeMember.EndsWith ("Method"):
+										memberKind = DynamicallyAccessedMemberKinds.PublicMethods;
+										break;
+
+									case var mt when getRuntimeMember.EndsWith ("Property"):
+										memberKind = DynamicallyAccessedMemberKinds.PublicProperties;
+										break;
+
+									default:
+										Debug.Fail ("Unsupported member type");
+										reflectionContext.RecordUnrecognizedPattern ($"Reflection call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' is of unexpected member type.");
```

Comment:
Nit: This should just throw - if we get here something very bad happened (the code is inconsistent).